app/src/main/java/de/westnordost/streetcomplete/quests/sport/AddSportForm.java

```diff
@@ -0,0 +1,150 @@
+			new SportValue("ice_hockey",			R.drawable.ic_sport_ice_hockey),
+			new SportValue("netball",				R.drawable.ic_sport_netball),
+			new SportValue("rugby",					R.drawable.ic_sport_rugby),
+			};
+
+	@Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
+									   Bundle savedInstanceState)
+	{
+		View view = super.onCreateView(inflater, container, savedInstanceState);
+
+		setTitle(R.string.quest_sport_title);
+
+		View contentView = setContentView(R.layout.quest_generic_list);
+
+		final RecyclerView sportList = (RecyclerView) contentView.findViewById(R.id.listSelect);
+		GridLayoutManager lm = new GridLayoutManager(getActivity(), 4);
+		sportList.setLayoutManager(lm);
+
+		final List<ImageSelectAdapter.Item> sportValuesList = Arrays.<ImageSelectAdapter.Item>asList(SPORTS_VALUES);
+
+		imageSelector = new ImageSelectAdapter();
+		imageSelector.setItems(sportValuesList.subList(0,MORE_THAN_95_PERCENT_COVERED));
+		if(savedInstanceState != null)
+		{
+			imageSelector.setSelectedIndex(savedInstanceState.getInt(SELECTED_INDEX, -1));
+		}
+		sportList.setAdapter(imageSelector);
+		sportList.setNestedScrollingEnabled(false);
+
+		final Button showMoreButton = (Button) contentView.findViewById(R.id.buttonShowMore);
+		showMoreButton.setOnClickListener(new View.OnClickListener()
+		{
+			@Override public void onClick(View v)
+			{
+				imageSelector.add(sportValuesList.subList(MORE_THAN_95_PERCENT_COVERED, sportValuesList.size()));
+				showMoreButton.setVisibility(View.GONE);
+			}
+		});
+
+		return view;
+	}
+
+	@Override public void onSaveInstanceState(Bundle outState)
+	{
+		super.onSaveInstanceState(outState);
+		outState.putInt(SELECTED_INDEX, imageSelector.getSelectedIndex());
+	}
+
+	@Override protected void onClickOk()
+	{
+		Bundle answer = new Bundle();
+		Integer selectedIndex = imageSelector.getSelectedIndex();
+		if(selectedIndex != -1)
+		{
+			String osmValue = SPORTS_VALUES[selectedIndex].osmValue;
+			answer.putString(SPORT, osmValue);
+		}
+		applyFormAnswer(answer);
+	}
+
+	@Override protected List<Integer> getOtherAnswerResourceIds()
+	{
+		List<Integer> answers = super.getOtherAnswerResourceIds();
+		answers.add(R.string.quest_sport_answer_multi);
+		return answers;
+	}
+
+	@Override protected boolean onClickOtherAnswer(int itemResourceId)
+	{
+		if(super.onClickOtherAnswer(itemResourceId)) return true;
+
+		if(itemResourceId == R.string.quest_sport_answer_multi)
+		{
+			Bundle answer = new Bundle();
+			answer.putString(SPORT, "multi");
+			applyImmediateAnswer(answer);
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override public boolean hasChanges()
+	{
+		return imageSelector.getSelectedIndex() != -1;
+	}
+
+	private static class SportValue extends ImageSelectAdapter.Item
+	{
+		public final String osmValue;
+
+		public SportValue(String osmValue, int drawableId)
+		{
+			super(drawableId, -1);
+			this.osmValue = osmValue;
+		}
+	}
+}
```

Comment:
There is quite some copy&paste code here from roof shape form. The first oppurtunity is always the best opportunity to refactor something.
Are you confident you can do that? Otherwise I would do this myself after the merge.